src/poetry/utils/dependency_specification.py

```diff
@@ -9,11 +9,13 @@
 from typing import TYPE_CHECKING
 from typing import Dict
 from typing import List
+from typing import TypeVar
 from typing import Union
 from typing import cast
 
 from poetry.core.packages.dependency import Dependency
 from poetry.core.packages.vcs_dependency import VCSDependency
+from tomlkit.items import InlineTable
```

Comment:
This is a little gross, but it's no worse than the tight coupling with logic in the factory we had before -- we can always revisit later.

Reply:
Yeah I am not a but fan of this one, but at present we need that for clarity.  Would have been better if tomlkit's typing was better. Let me know if you have any particular ideas to improve this in this PR.